Sources/ComposableArchitecture/SwiftUI/Alert.swift

```diff
@@ -128,47 +128,86 @@ public struct AlertState<Action> {
   }
 
   public struct Button {
-    public var action: Action?
-    public var animation: Animation??
-    public var type: `Type`
+    public var action: ButtonAction?
+    public var type: ButtonType
 
     public static func cancel(
       _ label: TextState,
-      send action: Action? = nil,
-      animation: Animation? = nil
+      action: ButtonAction? = nil
     ) -> Self {
-      Self(action: action, animation: animation, type: .cancel(label: label))
+      Self(action: action, type: .cancel(label: label))
     }
 
     public static func cancel(
-      send action: Action? = nil,
-      animation: Animation? = nil
+      action: ButtonAction? = nil
     ) -> Self {
-      Self(action: action, animation: animation, type: .cancel(label: nil))
+      Self(action: action, type: .cancel(label: nil))
     }
 
     public static func `default`(
       _ label: TextState,
-      send action: Action? = nil,
-      animation: Animation? = nil
+      action: ButtonAction? = nil
     ) -> Self {
-      Self(action: action, animation: animation, type: .default(label: label))
+      Self(action: action, type: .default(label: label))
     }
 
     public static func destructive(
       _ label: TextState,
-      send action: Action? = nil,
-      animation: Animation? = nil
+      action: ButtonAction? = nil
     ) -> Self {
-      Self(action: action, animation: animation, type: .destructive(label: label))
+      Self(action: action, type: .destructive(label: label))
     }
 
-    public enum `Type` {
-      case cancel(label: TextState?)
-      case `default`(label: TextState)
-      case destructive(label: TextState)
+    public static func cancel(
+      _ label: TextState,
+      send action: Action?
```

Comment:
Should I also deprecate these old ones? The one issue is we can't do a simple `rename` here...

Reply:
yeah I think deprecate. bummer we can't rename, but I think that's ok.